core/transaction_pool.go

```diff
@@ -180,30 +188,29 @@ func (pool *TxPool) SetLocal(tx *types.Transaction) {
 // validateTx checks whether a transaction is valid according
 // to the consensus rules.
 func (pool *TxPool) validateTx(tx *types.Transaction) error {
-	// Validate sender
-	var (
-		from common.Address
-		err  error
-	)
-
 	local := pool.localTx.contains(tx.Hash())
 	// Drop transactions under our own minimal accepted gas price
 	if !local && pool.minGasPrice.Cmp(tx.GasPrice()) > 0 {
 		return ErrCheap
 	}
 
-	// Validate the transaction sender and it's sig. Throw
-	// if the from fields is invalid.
-	if from, err = tx.From(); err != nil {
+	currentState, err := pool.currentState()
+	if err != nil {
+		return err
+	}
+
+	var from common.Address
+	if pool.homestead {
+		from, err = tx.From()
+	} else {
+		from, err = tx.FromFrontier()
+	}
+	if err != nil {
```

Comment:
If there aren't Frontier specific (i.e. homestead-invalid) transactions being bounced around currently, then I'm all for it. As long as we don't accidentally screw someone over we can restrict it straight away.Cheating a bit maybe... buut :)